Update SimpleBook plugin to allow authentication

slifty opened the pull request:
This PR updates the PHP portion of the SimpleBook plugin to support (1) authentication and (2) passthrough data.

It also updates the rendering server to pass those values to mw2pdf, although mw2pdf does not currently do anything with the passthrough data.

Related to issue #9

slifty committed:
Support SimpleBook authentication

Authentication is being handled similar to the way the original
Collection plugin worked: the MediaWiki instance calls the pdf rendering
service, passes credentials to use when rendering, and those credentials
are used.

We're making two changes:

1. Instead of using string manipulation to pass those credentials, we
   are now going to have those credentials passed as individual fields.

2. There is now support for "passthrough" data, which is a json string
   that gets converted to request parameters on individual render calls.

The goal of the first change is just to make the format for
authentication data more explicit.

The goal of the second change is to allow individual mediawiki instances
to have more control; for instance, it might pass contextual information
around what pages / data should be rendered based on what the logged in
user has access to.

SimpleBook itself doesn't need to know or care about the nature of the
passthrough data, it just commits to forwarding it along.

Issue #9

slifty committed:
Use tabs in json and php

The previous json and php files in this repository used tabs instead of spaces,
so this updates the editorconfig to reflect that existing standard.

slifty committed:
Update rendering server to use auth

Now that mw2pdf no longer relies on cookie jars to handle
authentication we can pass authentication credentials directly.

This updates the rendering server to use the new SimpleBook parameters
for both authentication and passthrough data.

Issue #9

hawkinsw commented on includes/rendering/MWPDFRenderingAPI.php:
```diff
@@ -7,23 +7,33 @@
+			$params['login_credentials'] = $wgSimpleBookRenderingCredentials;
+		}
+
+		if ( $wgSimpleBookRenderingCredentials ) {
```
This looks suspicious? Should it be `wgSimpleBookPassthroughParameters`?

slifty replied:
excellent catch thank you!

hawkinsw commented on services/api/app.py:
```diff
@@ -34,10 +33,13 @@ def print_to_pdf(urls, title, subtitle):
 
 def render_book():
```
This is just a question because I am stupid (ignorant, naive, etc) and had trouble following this function the first time that I saw it. 

Could we pass parameters to this function rather than relying on the global `request`? Ie, could we create some locals  from the `request` object in the route endpoint and pass those to `render_book`? I had a hard time grokking where `request` came from outside a function that has an obvious connection to an HTTP request. 

Thanks!!

slifty replied:
Great point!

This method existed before this PR so (if you don't mind!) I'd like to turn this observation into a new issue so it can be patched in a separate PR.

slifty replied:
Great point -- I was originally going to suggest we do this in another PR but realizing that I've upped the reliance on request in this PR to the point that it makes a lot of sense to just do it here and now.

slifty committed:
Make parameters optional

Authentication and passthrough parameters are not going to be needed by
every SimpleBook installation. This adjusts the rendering server so that
it will not pass those optional parameters if they don't exist.

Issue #9

slifty commented:
Fixed the items from the first review and also made an adjustment based on some of the discussion in the original issue (#9) related to the optional nature of authentication.  If auth and passthrough data is not provided as part of the request, those parameters will not be populated in the mw2pdf call.
